pilot/cmd/pilot-agent/main.go

```diff
@@ -223,6 +224,28 @@ var (
 			if err != nil {
 				return err
 			}
+			tokenManager := tokenmanager.CreateTokenManager(tokenManagerPlugin,
```

Comment:
Only if the flag is enabled ?

Reply:
tokenManager is also used by sts server, which is needed for token exchange for telemetry. Therefore, tokenManager is created regardless of the flag.

Reply:
STS server is also not enabled by default in OSS AFAIK - so it still needs to be enabled only if STS or gcp auth are enabled. I think we should add a comment that tokenmanager is really gcp token manager.

Reply:
Added the checking "if (stsPort > 0 || xdsAuthProvider != "")".

Reply:
In addition, added "a comment that tokenmanager is really gcp token manager."